### ISSUE TITLE
Automatically add new courses I am enrolled in

### ISSUE THREAD

**Issue by Aunib** [C0D3D3V/Moodle-Downloader-2#139]
## Avoid duplicates
* [X] This enhancement request has not already been raised before

## Is your feature request related to a problem? Please describe.
I wanted to look at a item in a moodle course I am enrolled in for some time, but sadly the downloader didn't download the item when it was accessible to me. This was because I did the moodle-dl setup before I was enrolled in this course and the course was not in the list of courses that moodle-dl downloads.

## Describe the solution you'd like
I would like an option that automatically adds new courses I enroll in or I am enrolled in to the list of courses that are downloaded every time my cronjob runs.

## Describe alternatives you've considered
none

## Additional context
none

**Comment by C0D3D3V** [comment]
tl;dr: You could use the `dont_download_course_ids` settings instead of the `download_course_ids` setting. 

That would download all courses down except for the excluded courses. Only as soon as you use the integrated configurator/wizard this is rewritten to download_course_ids, I thought at the time, I introduced it, that would be a good idea because you don't necessarily want to download every new course you have enrolled in online. And since you know if you have just enrolled, you can also change the setting quickly.


Autmoatically adding new courses to download_course_ids is not so easy with the current concept. Then we would have to build it the other way around and prefer the dont_download_course_ids setting instead of download_course_ids. 

First a workaround for now: Basically all courses will be downloaded as long as no ID is set in download_course_ids, so as long as you only use dont_download_course_ids all new courses will be downloaded.

**Comment by C0D3D3V** [comment]
We could add a question in the configurator if courses you enroll in in the future should be downloaded automatically and use `dont_download_course_ids` or `download_course_ids` depending on the answer.

**Comment by Aunib** [comment]
Thank you for the workaround, it seems to work (empty `dont_download_course_ids` seems to download all courses I am enrolled in.)
The solution you proposed sounds like a better solution to my problem than the one I proposed. I would like it if it would be implemented.

**Comment by C0D3D3V** [comment]
question added to the configurator in the new release
